nistats/tests/unit_tests/test_verify_events_file_uses_tab_separators.py

```diff
@@ -0,0 +1,147 @@
+        tmp_file_prefix, temp_file_suffix = (
+            'tmp ', ' ' + delimiter_name + '.csv')
+        with NamedTemporaryFile(mode='w', dir=os.getcwd(),
+                                prefix=tmp_file_prefix,
+                                suffix=temp_file_suffix) as temp_csv_obj:
+            _create_test_file(temp_csv=temp_csv_obj,
+                              test_data=data_for_temp_datafile,
+                              delimiter=delimiter_char)
+            _run_test_for_invalid_separator(filepath=temp_csv_obj.name,
+                                            delimiter_name=delimiter_name)
+
+
+def test_with_2D_dataframe():
+    data_for_pandas_dataframe, _ = make_data_for_test_runs()
+    events_pandas_dataframe = pd.DataFrame(data_for_pandas_dataframe)
+    result = _verify_events_file_uses_tab_separators(
+            events_files=events_pandas_dataframe)
+    expected_error = result[0][1]
+    with assert_raises(TypeError):
+        raise expected_error
+
+
+def test_with_1D_dataframe():
+    data_for_pandas_dataframe, _ = make_data_for_test_runs()
+    for dataframe_ in data_for_pandas_dataframe:
+        events_pandas_dataframe = pd.DataFrame(dataframe_)
+        result = _verify_events_file_uses_tab_separators(
+                events_files=events_pandas_dataframe)
+        expected_error = result[0][1]
+        with assert_raises(TypeError):
+            raise expected_error
+
+
+def test_for_invalid_filepath():
+    filepath = 'junk_file_path.csv'
+    result = _verify_events_file_uses_tab_separators(events_files=filepath)
+    expected_error = result[0][1]
+    with assert_raises(IOError):
+        raise expected_error
+
+
+def test_for_pandas_dataframe():
+    events_pandas_dataframe = pd.DataFrame([['a', 'b', 'c'], [0, 1, 2]])
+    result = _verify_events_file_uses_tab_separators(
+            events_files=events_pandas_dataframe)
+    expected_error = result[0][1]
+    with assert_raises(TypeError):
+        raise expected_error
+
+
+def test_binary_opening_an_image():
+    img_data = bytearray(
+            b'GIF87a\x01\x00\x01\x00\xe7*\x00\x00\x00\x00\x01\x01\x01\x02\x02'
+            b'\x07\x08\x08\x08\t\t\t\n\n\n\x0b\x0b\x0b\x0c\x0c\x0c\r;')
+    with NamedTemporaryFile(mode='wb', suffix='.gif',
+                            dir=os.getcwd()) as temp_img_obj:
+        temp_img_obj.write(img_data)
+        with assert_raises(ValueError):
+            _verify_events_file_uses_tab_separators(
+                    events_files=temp_img_obj.name)
+
+
+def test_binary_bytearray_of_ints_data():
+    temp_data_bytearray_from_ints = bytearray([0, 1, 0, 11, 10])
+    with NamedTemporaryFile(mode='wb', dir=os.getcwd(),
+                            suffix='.bin') as temp_bin_obj:
+        temp_bin_obj.write(temp_data_bytearray_from_ints)
+        with assert_raises(ValueError):
+            result = _verify_events_file_uses_tab_separators(
+                    events_files=temp_bin_obj.name)
+
+
+if __name__ == '__main__':
```

Comment:
I'm not familiar with pytests, but is this really necessary ? normally, the tests are run even without the main file ...

Reply:
This enables me to run the tests without invoking the frameworks, which is useful when I want to debug the tests, since I can set regular breakpoints like in standard python (not having to run it as a framework).
It also helps with debugging if something breaks in future . Running them only using the testing framework precludes this ability.
I can remove it, once the tests have been written successfully, or I can add a comment explaining its purpose. Either way it won't interfere with the code and is not difficult to understand.
What would you like?

Reply:
Makes sense, but I think that we should organize tests uniformally across modules (and is ppossible, with Nilearn).
For the time being, we can keep it as is -- it certainly does not harm.